### ISSUE TITLE
DAOS: scale - some clients fail to reconnect after failed connection 

### ISSUE THREAD

**Issue by frostedcmos** [ofiwg/libfabric#7228]
In the scenario of 1 server, 16 endpoints, 50clients, 32ppn (1600 total clients) some clients fail to connect to one of servers endpoints.

From logs we see the following situation.
Client_A attempts for the first time to connect to the endpoint=5 on the server.
Client_A does not get response and times out after 60 seconds; CART layer cancels this rpc and retries.
All retires of Client_A to connect to endpoint=5 still fail until test itself times out after 8+ minutes.
At the same time Client_A is able to connect to endpoint=0 on the same server.
Further other clients (Client_B) running on the same node as Client_A is able to talk to endpoint=5.

This was tried with latest ofi v1.14.0rc2

Reduced log from EDV run showing situation described above:

>> Failed communication with remote endpoint=5
08:01:36.93 eca061 DAOS[366879] rpc 0x1017010 rank 0 tag 5
08:02:36.93 eca061 DAOS[366879] ERR: rpc 0x1017010 timed out (60 seconds), target (0:5)
 
>> Successful communication with remote endpoint=5 by a different client
08:02:23.75 eca061 DAOS[366873] rpc 0x1bdf770 rank 0 tag 5
08:02:23.76 eca061 DAOS[366873] rpc 0x1bdf770 completed, dt_result 0.

>> Successful communication to remote endpoint=0
08:02:36.93 eca061 DAOS[366879] pool map_refresh() 01640108: 0x1024af0: asking rank 0 for version > 1
08:02:36.94 eca061 DAOS[366879] pool map_refresh_cb() 01640108: 0x1024af0: crt: DER_SUCCESS(0): 'Success'

>> Failed communication to remote endpoint=5
08:02:36.94 eca061 DAOS[366879] rpc 0x1017010rank 0 tag 5
08:03:36.94 eca061 DAOS[366879] ERR: rpc 0x1017010 timed out (60 seconds), target (0:5)

**Comment by shefty** [comment]
Notes based on separate chat thread.

A process is failing to send data to one of its AV entries.  The failure is that the underlying connection is never established.  Retrying the connection never succeeds.  Best data indicates that the IB CM returns a failure code of 28, which maps to consumer reject in IB CM terms.  That failure code can be generated by the ib_cm kernel module if the cm_id, which references the connection attempt, is destroyed by the app.  (We don't know that this is what is happening.)

We will need to add more log output to libfabric to determine if the connection request ever makes it to user space (DAOS server), and if so, why it was destroyed.

**Comment by shefty** [comment]
This was the error reported on one of the clients:
```
11728 libfabric:274624:ofi_rxm:ep_ctrl:rxm_handle_error():630<warn> fi_eq_readerr: err: Connection refused (111), prov_err: Unknown error -28 (-28)^
```
Trying to reproduce and obtain corresponding output from the server.

**Comment by shefty** [comment]
Working theory has the following patch as a fix:

https://github.com/shefty/libfabric/commit/750347bbad42a36014a346584621333e869da9d7

We can verify that this is the problem if we can see the log message (old or new) on the server when the client fails to connect.  So far, we haven't been able to get log messages out of the server beyond the initial fi_getinfo messages.

**Comment by shefty** [comment]
Reported problem should be resolved by PR #7239 (merged)

**Comment by shefty** [comment]
Now able to scale beyond reported issue.
